packages/integration-tests/suites/public-api/withScope/init.js

```diff
@@ -0,0 +1,11 @@
+import * as Sentry from '@sentry/browser';
+
+window.Sentry = Sentry;
+window.events = [];
+
+Sentry.init({
+  dsn: 'https://public@dsn.ingest.sentry.io/1337',
+  beforeSend: event => {
+    window.events.push(event);
```

Comment:
Why do we need this `beforeSend`?

Reply:
When testing with multiple events, we need to back up copies, as we may be testing them after they're sent and deleted.
This approach is [used in the old integration tests ](https://github.com/getsentry/sentry-javascript/blob/master/packages/browser/test/integration/common/init.js#L30), but alternatively I have implemented a helper function [there](https://github.com/getsentry/sentry-javascript/pull/4343/commits/7a3174a31ce882691c843f6e284c4ecfbf69f953) to keep track of multiple requests. It was intended for more complex cases, but we may switch to that everywhere.